File: frontend/src/features/admin/api/unit_catalog.ts
import { apiFetch } from '../../../lib/api'

export type UnitType = 'weight' | 'length' | 'volume' | 'count' | 'package'

export interface UnitCatalogOut {
  id: string
  code: string
  name: string
  symbol: string
  unit_type: UnitType
  is_active: boolean
}

export interface UnitCatalogCreate {
  id: string
  code: string
  name: string
  symbol: string
  unit_type: UnitType
}

export interface UnitCatalogUpdate {
  name?: string
  symbol?: string
  unit_type?: UnitType
  is_active?: boolean
}

export const UNIT_TYPE_LABELS: Record<UnitType, string> = {
  weight: 'Peso',
  length: 'Longitud',
  volume: 'Volumen',
  count: 'Cantidad',
  package: 'Empaque',
}

export function fetchUnitCatalog(onlyActive?: boolean): Promise<UnitCatalogOut[]> {
  const qs = onlyActive ? '?active_only=true' : ''
  return apiFetch<UnitCatalogOut[]>(`/units${qs}`)
}

export function createUnitCatalog(data: UnitCatalogCreate): Promise<UnitCatalogOut> {
  return apiFetch<UnitCatalogOut>('/units', { method: 'POST', body: JSON.stringify(data) })
}

export function updateUnitCatalog(id: string, data: UnitCatalogUpdate): Promise<UnitCatalogOut> {
  return apiFetch<UnitCatalogOut>(`/units/${id}`, { method: 'PATCH', body: JSON.stringify(data) })
}

export function deleteUnitCatalog(id: string): Promise<void> {
  return apiFetch<void>(`/units/${id}`, { method: 'DELETE' })
}

File: backend/app/schemas/unit_catalog.py
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.enums import UnitType


class UnitCatalogOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    code: str
    name: str
    symbol: str
    unit_type: UnitType
    is_active: bool


class UnitCatalogCreate(BaseModel):
    id: UUID
    code: str = Field(..., max_length=20)
    name: str = Field(..., max_length=50)
    symbol: str = Field(..., max_length=10)
    unit_type: UnitType


class UnitCatalogUpdate(BaseModel):
    name: str | None = Field(None, max_length=50)
    symbol: str | None = Field(None, max_length=10)
    unit_type: UnitType | None = None
    is_active: bool | None = None
